sdk/storage/Azure.Storage.Common/src/Shared/StorageBearerTokenChallengeAuthorizationPolicy.cs

```diff
@@ -81,6 +81,14 @@ private async ValueTask<bool> AuthorizeRequestOnChallengeInternalAsync(HttpMessa
                 // tenantId should be the guid as seen in this example: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/oauth2/authorize
                 tenantId = new Uri(authUri).Segments[1].Trim('/');
 
+                if (message.Response.Headers.TryGetValue(HttpHeader.Names.WwwAuthenticate, out string wwwAuthenticate)
+                    && wwwAuthenticate.Contains(Constants.ResourceId))
+                {
+                    string scope = wwwAuthenticate.Split(new string[] { Constants.ResourceId + "=" }, StringSplitOptions.None)[1];
```

Comment:
In this method, you already know the wwwAuth header is there. I'd also recommend that you use `AuthorizationChallengeParser.GetChallengeParameterFromResponse` to parse, as it is super efficient.

Reply:
fixed.